pills/17-nixpkgs-overriding-packages.xml

```diff
@@ -51,23 +51,20 @@
     </para>
     <screen><xi:include href="./17/fix-function.txt" parse="text" /></screen>
     <para>
-      It's a function that accepts a function <literal>f</literal>, calls <literal>f result</literal> on the result just returned by <literal>f result</literal> and returns it. In other words it's <literal>f(f(f(....</literal>
+      It's a function that accepts a function <literal>f</literal> and returns a function that recurses over f's arguments. Why is this useful? Because it plays nicely with Nix' laziness: it turns any function that otherwise would happily return (at least some) of its arguments unevaluated, into a function that recursively evaluates its arguments until there is nothing left to evaluate. To fix ideas (pun intended!) consider:
```

Comment:
> returns a function that recurses over f's arguments

It does not necessarily return a function:

```
nix-repl> lib.fix (lib.const "foo")
"foo"
```

Reply:
Indeed. It does so just in case it is given a function. I think this is the case we're interested in anyway. I hope this is clear enough in the text.